packages/@rescript/runtime/Stdlib_DataView.resi

```diff
@@ -0,0 +1,422 @@
+
+See [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) on MDN.
+
+## Examples
+
+```rescript
+let dv = DataView.fromBufferToEnd(buffer, ~byteOffset=4)
+```
+*/
+@new external fromBufferToEnd: (Stdlib_ArrayBuffer.t, ~byteOffset: int) => t = "DataView"
+
+/**
+`fromBufferWithRange(buffer, ~byteOffset, ~length)` creates a DataView for a specific byte range.
+
+See [`DataView`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) on MDN.
+
+## Examples
+
+```rescript
+let dv = DataView.fromBufferWithRange(buffer, ~byteOffset=2, ~length=8)
+```
+*/
+@new external fromBufferWithRange: (Stdlib_ArrayBuffer.t, ~byteOffset: int, ~length: int) => t = "DataView"
+
+/**
+`buffer(dataView)` returns the underlying ArrayBuffer referenced by this DataView.
+
+See [`DataView.buffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/buffer) on MDN.
+
+## Examples
+
+```rescript
+let arrBuf = DataView.buffer(dv)
+```
+*/
+@get external buffer: t => Stdlib_ArrayBuffer.t = "buffer"
+
+/**
+`byteLength(dataView)` returns the length (in bytes) of this DataView.
+
+See [`DataView.byteLength`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/byteLength) on MDN.
+
+## Examples
+
+```rescript
+let len = DataView.byteLength(dv)
+```
+*/
+@get external byteLength: t => int = "byteLength"
+
+/**
+`byteOffset(dataView)` returns the byte offset of this DataView from the start of its ArrayBuffer.
+
+See [`DataView.byteOffset`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/byteOffset) on MDN.
+
+## Examples
+
+```rescript
+let offset = DataView.byteOffset(dv)
+```
+*/
+@get external byteOffset: t => int = "byteOffset"
+
+/**
+`getInt8(dataView, offset)` reads an 8-bit signed integer at the specified byte offset.
+
+See [`DataView.getInt8`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getInt8) on MDN.
+
+## Examples
+
+```rescript
+let v = DataView.getInt8(dv, 0)
+```
+*/
+@send external getInt8: (t, int, ~littleEndian: bool=?) => int = "getInt8"
+
+/**
+`getUint8(dataView, offset)` reads an 8-bit unsigned integer at the specified byte offset.
+
+See [`DataView.getUint8`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getUint8) on MDN.
+
+## Examples
+
+```rescript
+let v = DataView.getUint8(dv, 1)
+```
+*/
+@send external getUint8: (t, int, ~littleEndian: bool=?) => int = "getUint8"
+
+/**
+`getInt16(dataView, offset, ~littleEndian=?)` reads a 16-bit signed integer at the specified byte offset, with optional endianness.
+
+See [`DataView.getInt16`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getInt16) on MDN.
+
+## Examples
+
+```rescript
+let v1 = DataView.getInt16(dv, 2)
+let v2 = DataView.getInt16(dv, 2, ~littleEndian=true)
```

Comment:
Examples should include checks with `==` if possible. These will get turned into test assertions.